aspnetcore/performance/caching/response.md

```diff
@@ -1,92 +1,72 @@
 
-**private**
+The primary HTTP header used for caching is `Cache-Control`. See the [HTTP 1.1 Caching](https://tools.ietf.org/html/rfc7234#section-5.2) and [Cache-Control](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9) for more information. Common cache directives:
 
-   Indicates the response is intended for a single user and **must not** be cached by a shared cache. The response could still be cached in a private cache (for instance, by the user's browser).
-
-**no-cache**
-
-   Indicates the response **must not** be used by a cache to satisfy any subsequent request (without successful revalidation with the origin server).
-
-> [!NOTE]
-> **Response caching does not cache responses on the web server**. It differs from [output caching](http://www.asp.net/mvc/overview/older-versions-1/controllers-and-routing/improving-performance-with-output-caching-cs), which would cache responses in memory on the server in earlier versions of ASP.NET and ASP.NET MVC. Output caching middleware is planned to be added to ASP.NET Core in a future release.
+* [public](https://tools.ietf.org/html/rfc7234#section-5.2.2.5)
+* [private](https://tools.ietf.org/html/rfc7234#section-5.2.2.6)
+* [no-cache](https://tools.ietf.org/html/rfc7234#section-5.2.1.4)
+* [Pragma](https://tools.ietf.org/html/rfc7234#section-5.4)
+* [Vary](https://tools.ietf.org/html/rfc7231#section-7.1.4)
 
-Additional HTTP headers used for caching include `Pragma` and `Vary`, which are described below. Learn more about [Caching in HTTP from the specification](https://tools.ietf.org/html/rfc7234#section-3).
+Caching responses on the web server can be enabled by adding the [response caching middleware](https://github.com/aspnet/responsecaching). See [Response caching middleware](middleware.md) for more information.
 
 ## ResponseCache Attribute
 
-The `ResponseCacheAttribute` is used to specify how a controller action's headers should be set to control its cache behavior. The attribute has the following properties, all of which are optional unless otherwise noted.
+The [ResponseCacheAttribute](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute) specifies the parameters necessary for setting appropriate headers in response caching. See [ResponseCacheAttribute](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute)  for a description of the parameters.
 
-**Duration `int`**
+`VaryByQueryKeys string[]`: When set, the response caching middleware will vary the stored response by the given list of query keys. The middleware will serve the stored response only if the query keys of the request matches those of the original request that generated the stored response. Setting this property without adding the middleware will throw a runttime exception. 
```

Comment:
Also, I'm guessing this is the only propery that is explicitly called out here because it's not yet in the API docs? Otherwise its heading seems different from the following ones.